src/server/page_server.hpp

```diff
@@ -300,7 +315,12 @@ class page_server
 
     followee_connection_handler_uptr_t m_followee_conn;
     std::vector<follower_connection_handler_uptr_t> m_follower_conn_vec;
+
     std::mutex m_followee_conn_mutex;
+    std::mutex m_follower_conn_vec_mutex;
```

Comment:
Nit.
`m_followee_conn_mutex` and `m_follower_conn_vec_mutex` are used mutually exclusive - AFAICT.
One PS cannot be at the same time a follower and a followee.
So, they can be just one - `m_follower_or_followee_depending_on_context_mutex` (a mostly silly suggestion :) ).